fetcher/fedora.go

```diff
@@ -0,0 +1,417 @@
+		logSuppressed: true,
+	}
+	log15.Info("Fetch CVE-ID list from bugzilla.redhat.com", "URL", req.url)
+	body, err := fetchFileWithUA(req)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch CVE-ID list, err: %w", err)
+	}
+
+	var b bugzillaXML
+	if err = xml.Unmarshal(body, &b); err != nil {
+		return nil, xerrors.Errorf("Failed to unmarshal xml. url: %s, err: %w", req.url, err)
+	}
+
+	var reqs []fetchRequest
+	for _, v := range b.Blocked {
+		req := fetchRequest{
+			url:           fmt.Sprintf("https://bugzilla.redhat.com/show_bug.cgi?ctype=xml&id=%s", v),
+			concurrently:  true,
+			logSuppressed: true,
+		}
+		reqs = append(reqs, req)
+	}
+
+	results, err := fetchFeedFiles(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch CVE-IDs, err: %w", err)
+	}
+
+	var ids []string
+	for _, result := range results {
+		var b bugzillaXML
+		if err = xml.Unmarshal(result.Body, &b); err != nil {
+			return nil, xerrors.Errorf("Failed to unmarshal xml. url: %s, err: %w", req.url, err)
+		}
+
+		if b.Alias != "" {
+			ids = append(ids, b.Alias)
+		}
+	}
+
+	log15.Info(fmt.Sprintf("%d CVE-IDs fetched", len(ids)))
+	return ids, nil
+}
+
+func extractInfoFromRepoMd(results []FetchResult, rt string, mt mimeType) ([]fetchRequest, error) {
+	var updateInfoReqs []fetchRequest
+	for _, r := range results {
+		var repoMd RepoMd
+		if err := xml.NewDecoder(bytes.NewBuffer(r.Body)).Decode(&repoMd); err != nil {
+			return nil, xerrors.Errorf("Failed to decode repomd of version %s. err: %w", r.Target, err)
+		}
+
+		for _, repo := range repoMd.RepoList {
+			if repo.Type != rt {
+				continue
+			}
+			u, err := url.Parse(r.URL)
+			if err != nil {
+				return nil, xerrors.Errorf("Failed to parse URL in XML. err: %w", err)
+			}
+			u.Path = strings.Replace(u.Path, "repodata/repomd.xml", repo.Location.Href, 1)
+			req := fetchRequest{
+				url:          u.String(),
+				target:       r.Target,
+				mimeType:     mt,
+				concurrently: true,
+			}
+			updateInfoReqs = append(updateInfoReqs, req)
+			break
+		}
+	}
+	return updateInfoReqs, nil
+}
+
+func fetchModuleInfoFromKojiPkgs(url, fileName string) (FedoraModuleInfo, error) {
+	req, err := newKojiPkgsRequest(url, fileName)
+	if err != nil {
+		return FedoraModuleInfo{}, xerrors.Errorf("Failed to generate request to kojipkgs.fedoraproject.org, err: %w", err)
+	}
+	result, err := fetchFileWithUA(req)
+	if err != nil {
+		return FedoraModuleInfo{}, xerrors.Errorf("Failed to fetch from kojipkgs.fedoraproject.org, err: %w", err)
+	}
+	moduleYaml, err := parseModulesYamlFedora(result)
+	if err != nil {
+		return FedoraModuleInfo{}, xerrors.Errorf("Failed to parse module text, err: %w", err)
+	}
+	if yml, ok := moduleYaml[fileName]; ok {
+		return yml, nil
+	} else {
+		return FedoraModuleInfo{}, xerrors.New("Module not found in kojipkgs.fedoraproject.org")
+	}
+}
+
+func newKojiPkgsRequest(url, fileName string) (fetchRequest, error) {
```

Comment:
Is it difficult to use the actual arch being fetched as an argument, instead of guessing the arch from the url?

Reply:
I checked result of each request,  but I couldn't determine which arch the information is, from the content of the answers.